### ISSUE TITLE
Can't upload, "HTTPError: 500 Server Error: Internal Server Error"

### ISSUE THREAD

**Issue by mike-lawrence** [caseymrm/drivesink#4]
Can create folders just fine, just can't upload actual files:

```
$ python drivesink.py upload ~/Pictures Pictures/backup
INFO       drivesink:83 _make_child_folder Creating remote folder backup in Pictures
INFO       drivesink:33 upload_child_file Uploading /Users/mike/Pictures/temp.png to backup
Traceback (most recent call last):
  File "drivesink.py", line 268, in <module>
    main()
  File "drivesink.py", line 257, in main
    drivesink.upload(args.source, args.destination)
  File "drivesink.py", line 124, in upload
    current_dir.upload_child_file(filename, local_path, node)
  File "drivesink.py", line 52, in upload_child_file
    headers={"Content-Type": m.content_type}))
  File "drivesink.py", line 211, in request_content
    path % self._config()["contentUrl"], **kwargs)
  File "drivesink.py", line 234, in _request
    req.raise_for_status()
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 834, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error
```

**Comment by alamers** [comment]
Hi, yeah I encountered that one as well. I'll work on it tomorrow, I managed to get it working with curl so just have to figure out why it isn't working in python.

**Comment by alamers** [comment]
There're some other projects out there that suffer from a similar issue, e.g. https://github.com/cnbeining/Amazon-Cloud-Drive-Python-SDK.

I did some fixes to make it more similar to a working curl call, but haven't been able to get it working from python. And python is not really my expertise so I gues I'll resort to use curl for that part.

**Comment by caseymrm** [comment]
Whoops, wrong issue. I'll see if I can reproduce this.

**Comment by caseymrm** [comment]
I wasn't able to reproduce this, but I added some code to print the json error Amazon sends back- might be interesting to try it again and see if anything is being printed.

**Comment by mike-lawrence** [comment]
Here's what I get with the latest version:
```
/usr/local/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
/usr/local/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
/usr/local/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
/usr/local/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
INFO       drivesink:34 upload_child_file Uploading /Users/mike/Pictures/temp.png to Pictures
/usr/local/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
Traceback (most recent call last):
  File "drivesink.py", line 287, in <module>
    main()
  File "drivesink.py", line 276, in main
    drivesink.upload(args.source, args.destination)
  File "drivesink.py", line 125, in upload
    current_dir.upload_child_file(filename, local_path, node)
  File "drivesink.py", line 53, in upload_child_file
    headers={"Content-Type": m.content_type}))
  File "drivesink.py", line 214, in request_content
    path % self._config()["contentUrl"], **kwargs)
  File "drivesink.py", line 253, in _request
    req.raise_for_status()
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 834, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error
```

**Comment by caseymrm** [comment]
Uploading works fine for me on any file I try.

Probably most useful would be to use Charles (http://www.charlesproxy.com/) to intercept the SSL calls for both drivesink and curl and see what the differences are. (It may be necessary to undo https://github.com/caseymrm/drivesink/commit/9744180c26b9f4f1b318a9f7b5f6f70e7f1e4df1 to get the SSL proxying working).

If anybody could do that, that would be really helpful!

**Comment by Sciruss** [comment]
According to this site https://developer.amazon.com/public/apis/experience/cloud-drive/content/restful-api-response-codes error 500 is "Servers are not working as expected. The request is probably valid but needs to be requested again later."

Is if possible to implement a retry if responses 500 or 503 are received?

**Comment by Sciruss** [comment]
Ferran Munoz has implemented Pagination as well as error 500 retries here, resolving two issues: https://github.com/ferranmunoz/drivesink/commit/3901f598555194976080e4a4f172be91038632aa

**Comment by caseymrm** [comment]
Thanks @Sciruss, merged with #25
